main.cpp

```diff
@@ -163,98 +171,414 @@ good to go!
+    DoubleType& divide( const IntType& it );
 };
+
 struct IntType
 {
-    int add(int lhs, int rhs);
-    int subtract(int lhs, int rhs);
-    int multiply(int lhs, int rhs);
-    int divide(int lhs, int rhs);
+    int* valuePtr;
+    IntType(int ivalue) : valuePtr(new int(ivalue)) {}
+    ~IntType(){delete valuePtr; valuePtr = nullptr;}
+
+    IntType& add(int rhs );
+    IntType& subtract(int rhs );
+    IntType& multiply(int rhs );
+    IntType& divide(int rhs );
+    
+    IntType& add( const FloatType& ft );
+    IntType& subtract( const FloatType& ft );
+    IntType& multiply( const FloatType& ft );
+    IntType& divide( const FloatType& ft );
+    
+    IntType& add( const DoubleType& dt );
+    IntType& subtract( const DoubleType& dt );
+    IntType& multiply( const DoubleType& dt );
+    IntType& divide( const DoubleType& dt );
+    
+    IntType& add( const IntType& it );
+    IntType& subtract( const IntType& it );
+    IntType& multiply( const IntType& it );
+    IntType& divide( const IntType& it );
 };
 
-float FloatType::add(float lhs, float rhs)
+FloatType& FloatType::add(float rhs)
+{
+    *valuePtr += rhs;
+    return *this;
+}
+
+FloatType& FloatType::subtract(float rhs)
+{
+    *valuePtr -= rhs;
+    return *this;
+}
+
+ FloatType& FloatType::multiply(float rhs)
+{
+    *valuePtr *= rhs;
+    return *this;
+}
+
+FloatType& FloatType::divide(float rhs)
+{
+    *valuePtr /= rhs;
+    return *this;
+}
+
+FloatType& FloatType::add(const FloatType& rhs)
+{
+    return add(*rhs.valuePtr);
+}
+
+FloatType& FloatType::subtract(const FloatType& rhs)
+{
+     return subtract(*rhs.valuePtr);
+}
+
+ FloatType& FloatType::multiply(const FloatType& rhs)
+{
+    return multiply(*rhs.valuePtr);
+}
+
+FloatType& FloatType::divide(const FloatType& rhs)
+{
+    return divide(*rhs.valuePtr);
+}
+
+FloatType& FloatType::add(const DoubleType& rhs)
+{
+    return add(*rhs.valuePtr);
+}
+
+FloatType& FloatType::subtract(const DoubleType& rhs)
+{
+    return subtract(*rhs.valuePtr);
+}
+
+ FloatType& FloatType::multiply(const DoubleType& rhs)
+{
+    return multiply(*rhs.valuePtr);
+}
+
+FloatType& FloatType::divide(const DoubleType& rhs)
+{
+    return divide(*rhs.valuePtr);
+}
+
+FloatType& FloatType::add(const IntType& rhs)
+{
+    return add(*rhs.valuePtr);
+}
+
+FloatType& FloatType::subtract(const IntType& rhs)
+{
+    return subtract(*rhs.valuePtr);
+}
+
+ FloatType& FloatType::multiply(const IntType& rhs)
+{
+    return multiply(*rhs.valuePtr);
+}
+
+FloatType& FloatType::divide(const IntType& rhs)
+{
+    return divide(*rhs.valuePtr);
+}
+
+DoubleType& DoubleType::add(double rhs)
+{
+    *valuePtr += rhs;
+    return *this;
+}
+
+DoubleType& DoubleType::subtract(double rhs)
+{
+    *valuePtr -= rhs;
+    return *this;
+}
+
+ DoubleType& DoubleType::multiply(double rhs)
+{
+    *valuePtr *= rhs;
+    return *this;
+}
+
+DoubleType& DoubleType::divide(double rhs)
+{
+    *valuePtr /= rhs;
+    return *this;
+}
```

Comment:
The previous project version issued a warning here.  Where is your warning?
- [ ] revise